### DISCUSSION TITLE
Flushing and Data Persistence

### DISCUSSION THREAD

**Discussion by dutysloth** [milvus-io/milvus#32996] [Q&A and General discussion] [answered]
Hi, just a simple question that probably has an answer somewhere but I couldn't find it -

If I insert entities into Milvus without flushing, and then say the whole Milvus cluster crashes.
Will I lose some data which is still in growing segments?

**Comment by xiaofan-luan** [comment, marked as answer]
no. flushing move data from log to minio. 
even if you don't do manualy flush, data could still be recovered from kafka/pulsar
